packages/edge-worker/src/EdgeWorker.ts

```diff
@@ -125,6 +131,37 @@ export class EdgeWorker extends EventEmitter {
 			timeoutMs: 10000,
 		});
 
+		// Initialize repository router with dependencies
+		const repositoryRouterDeps: RepositoryRouterDeps = {
+			fetchIssueLabels: async (issueId: string) => {
+				// Find which Linear client to use - use first repository's client that has this issue
+				for (const [_repoId, linearClient] of this.linearClients) {
+					try {
+						const issue = await linearClient.issue(issueId);
+						return await this.fetchIssueLabels(issue);
+					} catch (_error) {}
+				}
+				return [];
+			},
```

Comment:
this feels wrong, the code inside repositoryrouter which depends on this should be updated. the workspace/org associated with the issue should be known, and so should it also know which linear client that relates to.